### DISCUSSION TITLE
[Bug]: Local env stopped working after Docker upgrade

### DISCUSSION THREAD

**Discussion by caseybaggz** [supabase/supabase#16502] [Questions]
I recently setup my local environment and everything was working fine until I upgraded Docker to v.4.22.0 which now has broken the `supabase start` command. 😢 

I've tried uninstalling/re-installing Docker & SB CLI and it still crashes after the "setting up edge functions" task so I cannot use my local env at all anymore... Would love not to be blocked by this and actually have a fix so I can ship my app. 😭

```bash
Applying migration 20230813182909_remote_commit.sql...
Seeding data supabase/seed.sql...
supabase_edge_runtime_virtual-dnd container logs:
Download https://deno.land/std@0.182.0/http/server.ts
Download https://deno.land/x/jose@v4.13.1/index.ts
Sending fatal alert BadCertificate
Failed to load module: "https://deno.land/x/jose@v4.13.1/index.ts" - error sending request for url (https://deno.land/x/jose@v4.13.1/index.ts): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer

Caused by:
    0: error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
    1: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
CPU time used: 15ms
Error: worker boot error
service not healthy: [supabase_edge_runtime_virtual-dnd]
Try rerunning the command with --debug to troubleshoot the error.
```

I have 0 edge functions and have never created any.

**Comment by Ziinc** [comment]
This does not seem to be an issue with docker, since only certain requests are not successful.  Any success on retry?

**Reply by caseybaggz** [reply]
Nope. Still same error...every time. 😞

**Reply by caseybaggz** [reply]
...so I guess I'm just never going to be able to use SB locally again? This seems like it should be solvable since it's a Supabase API that's crashing and doesn't work? Is there some meta files or something that could potentially be cached on my machine that I maybe don't know about that SB uses?

**Reply by caseybaggz** [reply]
New error logs @Ziinc - same result. 😢 

```bash
1.4.0: Pulling from supabase/logflare
toomanyrequests: Rate exceeded
Retrying after 4s: public.ecr.aws/supabase/logflare:1.4.0
1.4.0: Pulling from supabase/logflare
c158987b0551: Already exists
fcacab8c2155: Pull complete
ef2cb6771e68: Pull complete
dd14f3d6237a: Pull complete
758eff7d0201: Pull complete
39fe5f6efb4e: Pull complete
a93bb64b00ca: Pull complete
4f4fb700ef54: Pull complete
3c2cc544d05b: Pull complete
Digest: sha256:e693c787ffe1ae17b6e4e920a3cdd212416d3e1f97e1bd7cb5b67de0abbb0264
Status: Downloaded newer image for public.ecr.aws/supabase/logflare:1.4.0
Applying migration 20230813182909_remote_commit.sql...
Seeding data supabase/seed.sql...
⣽ Setting up Edge Functions runtime...

Applying migration 20230813182909_remote_commit.sql...
Seeding data supabase/seed.sql...
supabase_edge_runtime_virtual-dnd container logs:
Download https://deno.land/std@0.182.0/http/server.ts
Download https://deno.land/x/jose@v4.13.1/index.ts
Sending fatal alert BadCertificate
CPU time used: 37ms
Failed to load module: "https://deno.land/x/jose@v4.13.1/index.ts" - error sending request for url (https://deno.land/x/jose@v4.13.1/index.ts): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer

Caused by:
    0: error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
    1: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
Error: worker boot error
service not healthy: [supabase_edge_runtime_virtual-dnd]
Try rerunning the command with --debug to troubleshoot the error.
 ELIFECYCLE  Command failed with exit code 1.
```

**Reply by Ziinc** [reply]
Seems like your network uses self-signed certificates, and deno rejects any self-signed certificates for security reasons.
 @laktek you have any idea on this?

**Reply by caseybaggz** [reply]
Interesting. Yeah I would love to learn more since it was working fine up until I literally upgraded from an ancient version of Docker to the latest version which is why I thought it was a Docker issue?

**Reply by laktek** [reply]
@caseybaggz Since you're not using edge functions, can you try running `supabase start --exclude edge-runtime`?

**Reply by caseybaggz** [reply]
@laktek great news is that - that worked! Now we just need to figure out why edge-runtime specially is crashing the build in regards to deno and how to fix it. It worked fine before, so there has to be a way?
